components/operator/internal/controllers/stack/delete/brokers.go

```diff
@@ -0,0 +1,68 @@
+package delete
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/formancehq/operator/apis/stack/v1beta3"
+	"github.com/formancehq/operator/internal/controllers/stack/storage/nats"
+	"github.com/go-logr/logr"
+)
+
+var (
+	natsClientId = "membership"
+)
+var (
+	ErrCast = errors.New("cannot cast interface to string")
+)
+
+func DeleteByBrokers(c *v1beta3.Configuration, stackName string, subjectService []string, logger logr.Logger) error {
+	values := reflect.ValueOf(c.Spec.Broker)
+	for i := 0; i < values.NumField(); i++ {
+		if rv := reflect.ValueOf(values.Field(i)); !rv.IsValid() || rv.IsNil() {
+			continue
+		}
+		switch t := values.Field(i).Interface().(type) {
+		case v1beta3.NatsConfig:
+			if err := deleleNatsSubjects(&t, stackName, subjectService, logger); err != nil {
+				return err
+			}
+		}
+
+	}
+	return nil
+}
+
+func deleleNatsSubjects(config *v1beta3.NatsConfig, stackName string, subjectService []string, logger logr.Logger) error {
+	client, err := nats.NewClient(config, natsClientId)
+	if err != nil {
+		logger.Error(err, "NATS: client")
+		return err
+	}
+	defer client.Close()
+
+	for _, service := range subjectService {
+		stackServiceSubject := fmt.Sprintf("%s-%s", stackName, service)
+
+		exist, err := nats.ExistSubject(client, stackServiceSubject, logger)
+		if err != nil {
+			logger.Error(err, "NATS: subject existancy check")
```

Comment:
Avoir writing too many logs if you return the error.
Prefer return the error, you can wrap it if you want, and display it at top level.